src/main/app/common/country.ts

```diff
@@ -0,0 +1,122 @@
+    return [
+      Country.WALES,
+      Country.ENGLAND
+    ]
+  }
+
+  static async isValidClaimantAddress (form: Form<PartyDetails>): Promise<Form<PartyDetails>> {
+    await this.getCountry(form.model.address.postcode).then((country: string) => {
+      if (country.length > 0 && !Country.isClaimantCountry(country)) {
+        let error = new ValidationError()
+        error.property = 'address[postcode]'
+        error.constraints = { amount: 'The country must be England, Wales, Scotland or Northern Ireland' }
+        form.errors.push(new FormValidationError(error))
+      }
+    })
+
+    if (form.model.hasCorrespondenceAddress) {
+      await this.getCountry(form.model.address.postcode).then((country: string) => {
+        if (country.length > 0 && !Country.isClaimantCountry(country)) {
+          let error = new ValidationError()
+          error.property = 'correspondenceAddress[postcode]'
+          error.constraints = { amount: 'The country must be England, Wales, Scotland or Northern Ireland' }
+          form.errors.push(new FormValidationError(error))
+        }
+      })
+    }
+
+    return form
+  }
+
+  static async isValidDefendantAddress (form: Form<PartyDetails>): Promise<Form<PartyDetails>> {
+    await this.getCountry(form.model.address.postcode).then((country: string) => {
+      if (country.length > 0 && !Country.isDefendantCountry(country)) {
+        let error = new ValidationError()
+        error.property = 'address[postcode]'
+        error.constraints = { amount: 'The country must be England or Wales' }
+        form.errors.push(new FormValidationError(error))
+      }
+    })
+
+    if (form.model.hasCorrespondenceAddress) {
+      await this.getCountry(form.model.address.postcode).then((country: string) => {
+        if (country.length > 0 && !Country.isDefendantCountry(country)) {
+          let error = new ValidationError()
+          error.property = 'correspondenceAddress[postcode]'
+          error.constraints = { amount: 'The country must be England or Wales' }
+          form.errors.push(new FormValidationError(error))
+        }
+      })
+    }
+
+    return form
+  }
+
+  static isClaimantCountry (value: string): boolean {
+    return Country.all().filter(country => country.name.toLowerCase() === value.toLowerCase()).length > 0
+  }
+
+  static isDefendantCountry (value: string): boolean {
+    return Country.defendantCountries().filter(country => country.name.toLowerCase() === value.toLowerCase()).length > 0
+  }
+
+  static async getCountry (postcode: string): Promise<string> {
+    if (postcode === undefined || postcode.length === 0) {
+      return ''
+    } else {
+      let country = ''
+      await postcodeClient.lookupPostcode(postcode)
+        .then((postcodeInfoResponse: PostcodeInfoResponse) => {
+          if (postcodeInfoResponse.valid) {
+            const object: PostcodeInfoCountry = postcodeInfoResponse.country as any as PostcodeInfoCountry
```

Comment:
`postcodeInfoResponse.country as any as PostcodeInfoCountry`?

Reply:
the response from the postcodelookup client is a JSON object so when I try to parse it is throws an error saying it's already JSON. when I try to just assign it to the variable TypeScript wont compile it as doesn't see it as a proper casting to an object so had to cast it to any to then cast it as a PostcodeInfoCountry